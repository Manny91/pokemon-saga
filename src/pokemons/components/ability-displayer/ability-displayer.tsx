import { PokemonAbility } from "../../../services/pokemon.service";
import React from "react";
import styled from "../../../styled-components";

interface AbilitiesDisplayerProps {
  abilities: PokemonAbility[];
}

export const PokemonAbilitiesDisplayer = ({
  abilities
}: AbilitiesDisplayerProps) => {
  const abilitiesFlatted = flatAbilities(abilities);
  return (
    abilitiesFlatted && (
      <AbilityWrapper>
        <PokeAbilityContainerHeader> Abilities: </PokeAbilityContainerHeader>
        {abilitiesFlatted.map((ability, i) => {
          return (
            <PokeAbilityContainer key={i}>{ability.name}</PokeAbilityContainer>
          );
        })}
      </AbilityWrapper>
    )
  );
};

function flatAbilities(abilities: PokemonAbility[]): Ability[] {
  const abilityFlat = abilities.map(ability => {
    return ability.ability;
  });
  return abilityFlat;
}

const AbilityWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: ${props => props.theme.colors.greenScreen};
  padding: 15px 20px;
  border-radius: 3px;
  flex-grow: 1;
  min-width: 120px;
  border: inset ${props => props.theme.colors.greenBorder} 3px;
`;

const PokeAbilityContainer = styled.p`
  margin: 5px 0px;
  font-weight: bold;
  text-transform: capitalize;
`;

const PokeAbilityContainerHeader = styled.h4`
  text-transform: capitalize;
`;
interface Ability {
  name: string;
}
