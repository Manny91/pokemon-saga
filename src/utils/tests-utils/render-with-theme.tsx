import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
