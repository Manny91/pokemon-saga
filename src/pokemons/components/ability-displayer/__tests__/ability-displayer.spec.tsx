import React from "react";
import { PokemonAbilitiesDisplayer } from "../ability-displayer";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonAbilitiesDisplayer Component", () => {
  it("should render properly", () => {
    const abilities = [
      { ability: { name: "test ability" } },
      { ability: { name: "test ability 2" } }
    ];
    const subject = renderWithTheme(
      <PokemonAbilitiesDisplayer abilities={abilities} />
    );
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
