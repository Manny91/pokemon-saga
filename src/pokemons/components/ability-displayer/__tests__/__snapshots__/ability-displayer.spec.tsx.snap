// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonAbilitiesDisplayer Component should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #A5CD53;
  padding: 15px 20px;
  border-radius: 3px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 120px;
  margin: 5px;
  border: inset #9aa28b 3px;
}

.c2 {
  margin: 5px 0px;
  font-weight: bold;
  text-transform: capitalize;
}

.c1 {
  text-transform: capitalize;
}

<div>
  <div
    class="c0"
  >
    <h4
      class="c1"
    >
       Abilities: 
    </h4>
    <p
      class="c2"
    >
      test ability
    </p>
    <p
      class="c2"
    >
      test ability 2
    </p>
  </div>
</div>
`;
